lookup-cars.ts

```diff
@@ -0,0 +1,117 @@
+import * as Link from 'multiformats/link'
+import { base58btc } from 'multiformats/bases/base58'
+import { DynamoDBClient, QueryCommand } from '@aws-sdk/client-dynamodb'
+import fs from 'fs'
+import readline from 'readline'
+import 'dotenv/config'
+
+const BLOCK_POSITION_TABLE = 'prod-ep-v1-blocks-cars-position'
+const MAX_SOCKETS = parseInt(process.env.STORACHA_DOTSTORAGE_LOOKUP_MAX_SOCKETS ?? '100')
+const BATCH_SIZE = parseInt(process.env.STORACHA_DOTSTORAGE_LOOKUP_BATCH_SIZE ?? '5000')
+
+const dynamodb = new DynamoDBClient({
+  region: 'us-west-2',
+  credentials: {
+    accessKeyId: process.env.AWS_ACCESS_KEY_ID!,
+    secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY!
+  },
+  requestHandler: {
+    httpsAgent: { maxSockets: MAX_SOCKETS },
+  }
+})
+
+// Cars CSV file path
+const carsPath = 'data/cars.csv'
+
+function parseContentCid (contentCid: string): string {
+  const link = Link.parse(contentCid)
+  return base58btc.encode(link.multihash.bytes)
+}
+
+async function lookupByBlockMultihash (blockmultihash: string): Promise<string[]> {
+  try {
+    const query = await dynamodb.send(new QueryCommand({
+      TableName: BLOCK_POSITION_TABLE,
+      KeyConditionExpression: 'blockmultihash = :blockmultihash',
+      ExpressionAttributeValues: {
+        ':blockmultihash': { S: blockmultihash }
+      },
+      ProjectionExpression: 'carpath'
+    }))
+    
+    if (query.Items && query.Items.length > 0) {
+      return query.Items
+        .map(item => item.carpath?.S)
+        .filter((carpath): carpath is string => carpath !== undefined)
+    }
+  } catch (error) {
+    console.error(`Error querying ${BLOCK_POSITION_TABLE} for ${blockmultihash}:`, error)
+  }
+  
+  return []
+}
+
+// thanks, https://stackoverflow.com/posts/66762031/revisions
+async function* batches<T> (iterable: AsyncIterable<T>, batchSize: number) {
+  let items: T[] = [];
+  for await (const item of iterable) {
+    items.push(item);
+    if (items.length >= batchSize) {
+      yield items;
+      items = []
+    }
+  }
+  if (items.length !== 0) {
+    yield items;
+  }
+}
+
+// thanks, https://stackoverflow.com/posts/55435856/revisions
+function* chunks<T> (arr: T[], n: number): Generator<T[], void> {
+  for (let i = 0; i < arr.length; i += n) {
+    yield arr.slice(i, i + n);
+  }
+}
+
+async function processLines (lines: Iterable<string>) {
+  for (const line of lines) {
+    const contentCid = line.trim()
+    if (!contentCid) continue // Skip empty lines
+
+    try {
+      const blockmultihash = parseContentCid(contentCid)
+      const carpaths = await lookupByBlockMultihash(blockmultihash)
+      
+      if (carpaths.length > 0) {
+        for (const carpath of carpaths) {
+          const csvLine = `${blockmultihash},${carpath}`
+          fs.appendFileSync(carsPath, `${csvLine}\n`)
```

Comment:
no attempt to pull out the ones that are in backup urls? Or is that a post processing step?

Reply:
yea imho a better job for unix - I want to get the raw data out of dynamo in this step, and then pill the car CIDs out to a separate file and use `comm -23 <(sort set1_file) <(sort set2_file) # Elements in set1_file but not in set2_file` to remove any car CIDs I already deleted in the last step